data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/controller/MultipartController.java

```diff
@@ -62,45 +61,46 @@
     private static final String PAYLOAD = "payload";
 
     private final Monitor monitor;
-    private final String connectorId;
     private final List<Handler> multipartHandlers;
     private final ObjectMapper objectMapper;
     private final IdentityService identityService;
+    private final IdsResponseMessageFactory responseMessageFactory;
 
     public MultipartController(@NotNull Monitor monitor,
-                               @NotNull String connectorId,
                                @NotNull ObjectMapper objectMapper,
                                @NotNull IdentityService identityService,
+                               @NotNull IdsResponseMessageFactory responseMessageFactory,
                                @NotNull List<Handler> multipartHandlers) {
         this.monitor = Objects.requireNonNull(monitor);
-        this.connectorId = Objects.requireNonNull(connectorId);
         this.objectMapper = Objects.requireNonNull(objectMapper);
         this.multipartHandlers = Objects.requireNonNull(multipartHandlers);
+        this.responseMessageFactory = Objects.requireNonNull(responseMessageFactory);
         this.identityService = Objects.requireNonNull(identityService);
     }
 
     @POST
     public Response request(@FormDataParam(HEADER) InputStream headerInputStream,
                             @FormDataParam(PAYLOAD) String payload) {
         if (headerInputStream == null) {
-            return Response.ok(createFormDataMultiPart(malformedMessage(null, connectorId))).build();
+            return Response.status(Response.Status.BAD_REQUEST).build();
         }
 
         Message header;
         try {
             header = objectMapper.readValue(headerInputStream, Message.class);
         } catch (IOException e) {
-            return Response.ok(createFormDataMultiPart(malformedMessage(null, connectorId))).build();
+            return Response.status(Response.Status.BAD_REQUEST).build();
         }
 
         if (header == null) {
-            return Response.ok(createFormDataMultiPart(malformedMessage(null, connectorId))).build();
+            return Response.status(Response.Status.BAD_REQUEST).build();
         }
 
         DynamicAttributeToken dynamicAttributeToken = header.getSecurityToken();
         if (dynamicAttributeToken == null || dynamicAttributeToken.getTokenValue() == null) {
             monitor.warning("MultipartController: Token is missing in header");
-            return Response.ok(createFormDataMultiPart(notAuthenticated(header, connectorId))).build();
+            RejectionMessage notAuthenticatedMessage = responseMessageFactory.createNotAuthenticatedMessage(header);
```

Comment:
The ResponseMessageFactory could throw an exception when creating the message.  InvalidCorrelationMessage- and MissingClientCredentialsExceptions should be caught here.